chain/client/src/info.rs

```diff
@@ -177,6 +181,45 @@ impl InfoHelper {
         }
     }
 
+    fn record_epoch_settlement_info(head: &Tip, client: &crate::client::Client) {
+        let epoch_info = client.runtime_adapter.get_epoch_info(&head.epoch_id);
+        let blocks_in_epoch = client.config.epoch_length;
+        let number_of_shards =
+            client.runtime_adapter.num_shards(&head.epoch_id).unwrap_or_default();
+        if let Ok(epoch_info) = epoch_info {
+            let (blocks, chunks) = client
+                .runtime_adapter
+                .get_epoch_start_height(&head.last_block_hash)
+                .map_or(0..0, |epoch_start_height| {
+                    epoch_start_height..(epoch_start_height + blocks_in_epoch)
+                })
+                .fold(
```

Comment:
I think this would be easier to read if we just change it to a regular for loop; the fold is only used to iterate over the indexes, not to, say, fold some function over the actual elements in the container.